client/client.go

```diff
@@ -83,6 +83,7 @@ func (this *Client) exhangeRefreshTokenForAccessToken(clientHostUrl string, refr
 	}
 	if resp.StatusCode != 200 {
 		err = errors.New(fmt.Sprintf("invalid status code %+v", resp))
+		return
 	}
 	defer resp.Body.Close()
```

Comment:
It might be safer to put this defer before checking the status code, because otherwise the client could return without closing the response body: https://stackoverflow.com/a/55210335